includes/class-wc-payment-gateway-wcpay.php

```diff
@@ -2311,23 +2312,21 @@ static function ( $refund ) use ( $refund_amount ) {
 			$currency = strtoupper( $refund['currency'] );
 			Tracker::track_admin( 'wcpay_edit_order_refund_success' );
 		} catch ( Exception $e ) {
-			if ( $e instanceof API_Exception && 'insufficient_balance_for_refund' === $e->get_error_code() ) {
-				// Handle insufficient_balance_for_refund error.
-				$this->order_service->handle_insufficient_balance_for_refund( $order, WC_Payments_Utils::prepare_amount( $amount, $order->get_currency() ) );
+			if ( $e instanceof API_Exception ) {
+				$failure_reason = $e->get_error_code();
```

Comment:
Based on https://docs.stripe.com/refunds#failed-refunds

> When using the API, a [Refund](https://docs.stripe.com/api#refund_object) object’s status transitions to failed and includes these attributes... 

In other words, `failure_reason` is only available in webhook events. 

For the sync request here, as well as [class-wc-rest-payments-refunds-controller.php](https://github.com/Automattic/woocommerce-payments/blob/e7058eec53c5a8890a9796e0d9de9abd0936723e/includes/admin/class-wc-rest-payments-refunds-controller.php#L72-L73), we can not get the failure reason. Therefore, we should keep the current logic.

Reply:
Thanks so much for catching this Dat! I don't know what I was thinking 🤦  I reverted this part in this commit [b1fdc52](https://github.com/Automattic/woocommerce-payments/pull/10677/commits/b1fdc52d75774956f0be73598fa1d3f16c433d94)